auth/service_test.go

```diff
@@ -630,3 +651,35 @@ func TestDBAuthService_DeleteUser(t *testing.T) {
 		t.Errorf("GetUser(%s) after deletion: %s", userName, err)
 	}
 }
+
+func BenchmarkDBAuthService_ListEffectivePolicies(b *testing.B) {
+	// setup user with policies for benchmark
+	adb, _ := testutil.GetDB(b, databaseURI)
+	service := auth.NewDBAuthService(adb, crypt.NewSecretStore(someSecret), authparams.ServiceCache{
+		Enabled: false,
+	})
+	userName := userWithPolicies(b, service, userPoliciesForTesting)
+	// test list effective policies with and without cache
+	b.Run("with_cache", func(b *testing.B) {
+		serviceWithCache := auth.NewDBAuthService(adb, crypt.NewSecretStore(someSecret), authparams.ServiceCache{
+			Enabled:        true,
+			Size:           1024,
+			TTL:            20 * time.Second,
```

Comment:
Not sure you're allowed to do this.  AFAIU `go bench` runs increasing `b.N` until the benchmark has run for "long enough" (`--benchtime`).  So you're _not_ allowed to change the time an iteration takes based on N.  So... if it tries too large an N, and runs for 17.2 seconds, it will automatically run somewhat slower.

See e.g. https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go, "traps for young players".

Either pick a _really_ short TTL (on the order of 500ms+-50ms) and measure the effects of expiry, or pick a _really_ long TTL (on the order of 1d) and measure the effects of a cache without expiry.

Reply:
1. I'm passing the benchmark object to `benchmarkListEffectivePolicies` and reset the timer - so the above code is just benchmark setup code, not part of the measure.
Using b.N for calling `ListEffectivePolicies`.
Missing where I miss-used the N. if I keep using the same parameters for the cache system.
I can move the service creation out of the Run call to make sure it is not part of the bench - but I am sure call reset should solve this one.
2. I'll add another test to try to simulate the expiry - wanted the good numbers

Reply:
add test with cache with low ttl value